osu.Game/Screens/Play/Player.cs

```diff
@@ -693,6 +695,9 @@ public override bool OnExiting(IScreen next)
 
             musicController.ResetTrackAdjustments();
 
+            // Ensure we reset the IsGameplay state
+            isGameplay.Value = false;
```

Comment:
If this is intended as a catch-all, it should probably be run in `OnSuspending` as well.

Hard to guarantee this behaviour is correct as there's no test coverage.

Reply:
I can add a test case for this, but it will only be verifiable in the test browser, since `OsuGame.IsGameplay` is not available headless.

Reply:
You can make `isGameplay` protected and test it directly.

Reply:
As in, it's defined in `OsuGame` but we don't use that headless.